### ISSUE TITLE
Moon biome _still_ in overworld

### ISSUE THREAD

**Issue by SephirothWS** [BBoldt/Unabridged#526]
**Minecraft version:** 1.7.10

**Unabridged version:** R13a

**Issue Description:** ![Image](https://i.imgur.com/d6blUsO.jpg)

**Steps to Reproduce:** 
1.  Load up the modpack.
2.  Click Singleplayer
3.  Click New World.
4.  Choose Game Type of Creative
5.  Select More World Options
6.  Set the seed to -732007273
7.  Enable Cheats
8.  Click Create World
9.  Wait for world to load.
10. Fly to the coordinates: **X**: -229, **Y**: 131, **Z**: 1169
11. ???
12. Profit.

**Crash report or log (uploaded to gist.github.com or pastebin.com):**
N/A

**Comment by BBoldt** [comment]
ugh

**Comment by bjornthv** [comment]
I've even had a world with the MARS biome in the overworld... I have no evidence atm, but as oon as I stumble upon it again, I will post.

**Comment by BBoldt** [comment]
Well I can't tell if its fixed, but that spot is different in the newest unreleased version of Unabridged III

![Image](https://i.imgur.com/dgQlVeB.jpg)

**Comment by Fuhrmaaj** [comment]
Heya, I just started a new world and I would say about 50% of the biomes are either conflicted, or they are the "-=|World-Engine|=-" biome. I did a test on an instance and it said that there are 33 biome ID conflicts and 2 potion ID conflicts. I got a concussion this week, but I'm going to see if I can't go over all the biome stuff this weekend.

*************************

Total amount of conflicted biome ids: 33
IDs in conflict:

0 is asked by: Ocean (net.minecraft.world.biome.BiomeGenOcean)
1 is asked by: Plains (net.minecraft.world.biome.BiomeGenPlains)
2 is asked by: Desert (net.minecraft.world.biome.BiomeGenDesert)
3 is asked by: Extreme Hills (net.minecraft.world.biome.BiomeGenHills)
4 is asked by: Forest (net.minecraft.world.biome.BiomeGenForest)
5 is asked by: Taiga (net.minecraft.world.biome.BiomeGenTaiga)
6 is asked by: Swampland (net.minecraft.world.biome.BiomeGenSwamp)
7 is asked by: River (net.minecraft.world.biome.BiomeGenRiver)
8 is asked by: Hell (net.minecraft.world.biome.BiomeGenHell), Hell (biomesoplenty.common.biome.overridden.BiomeGenBOPHell)
12 is asked by: Ice Plains (net.minecraft.world.biome.BiomeGenSnow)
14 is asked by: MushroomIsland (net.minecraft.world.biome.BiomeGenMushroomIsland)
15 is asked by: MushroomIslandShore (net.minecraft.world.biome.BiomeGenMushroomIsland)
17 is asked by: DesertHills (net.minecraft.world.biome.BiomeGenDesert)
18 is asked by: ForestHills (net.minecraft.world.biome.BiomeGenForest)
19 is asked by: TaigaHills (net.minecraft.world.biome.BiomeGenTaiga)
20 is asked by: Extreme Hills Edge (net.minecraft.world.biome.BiomeGenHills)
21 is asked by: Jungle (net.minecraft.world.biome.BiomeGenJungle)
22 is asked by: JungleHills (net.minecraft.world.biome.BiomeGenJungle)
23 is asked by: JungleEdge (net.minecraft.world.biome.BiomeGenJungle)
27 is asked by: Birch Forest (net.minecraft.world.biome.BiomeGenForest)
28 is asked by: Birch Forest Hills (net.minecraft.world.biome.BiomeGenForest)
29 is asked by: Roofed Forest (net.minecraft.world.biome.BiomeGenForest)
30 is asked by: Cold Taiga (net.minecraft.world.biome.BiomeGenTaiga)
31 is asked by: Cold Taiga Hills (net.minecraft.world.biome.BiomeGenTaiga)
35 is asked by: Savanna (net.minecraft.world.biome.BiomeGenSavanna)
36 is asked by: Savanna Plateau (net.minecraft.world.biome.BiomeGenSavanna)
37 is asked by: Mesa (net.minecraft.world.biome.BiomeGenMesa)
38 is asked by: Mesa Plateau F (net.minecraft.world.biome.BiomeGenMesa)
39 is asked by: Mesa Plateau (net.minecraft.world.biome.BiomeGenMesa)
102 is asked by: Lush Swamp (biomesoplenty.common.biome.overworld.BiomeGenLushSwamp), moon (micdoodle8.mods.galacticraft.core.world.gen.BiomeGenFlatMoon)
103 is asked by: Mangrove (biomesoplenty.common.biome.overworld.sub.BiomeGenMangrove), marsFlat (micdoodle8.mods.galacticraft.planets.mars.world.gen.BiomeGenFlatMars)
105 is asked by: Marsh (biomesoplenty.common.biome.overworld.BiomeGenMarsh), space (micdoodle8.mods.galacticraft.core.world.gen.BiomeGenBaseOrbit)
160 is asked by: Mega Spruce Taiga (net.minecraft.world.biome.BiomeGenTaiga), Maple Woods (biomesoplenty.common.biome.overworld.BiomeGenMapleWoods)

***************************

Total amount of conflicted potion ids: 2
IDs in conflict:

29 is asked by: potion.radiation (galaxyspace.core.prefab.potions.Radiation), immersiveengineering.potion.conductive (blusunrize.immersiveengineering.common.util.IEPotions$IEPotion)
30 is asked by: potion.antiradiation (galaxyspace.core.prefab.potions.AntiRadiation), immersiveengineering.potion.sticky (blusunrize.immersiveengineering.common.util.IEPotions$IEPotion)

**Comment by Fuhrmaaj** [comment]
OK, so I've figured out that the biomes 0-39 (with some omissions) are all mentioned in the Biomes O' Plenty config files. In biomegen.cfg, there is a list of "vanilla biomes to override" which exactly matches up with the early biomes. The only one which is set to "true" is Hell, which is why both vanilla and BoP are mentioned for ID#8. My best guess is that RWG overwrites those biomes as well, but there isn't a config option to disable that.

I forgot to make my world spawn as RWG, which might explain why so many biomes are screwed up then. If we could figure out how to set up a note that the mod authors intend for you to use RWG when generating world, and maybe a little warning when you chose the wrong option, then that's probably the best solution. I'm going to try to dig into that, and also test that the RWG world does work.

The other 4 are clearly just an ID mistake. I'll PR the fixes when I have them.

Edit: I used the Anti Id Conflict mod to generate the list from the above post. I noticed after that it didn't list the Tropicraft biomes, so I went through and noticed that the GalaxySpace biomes are also not listed. I don't know how best to handle this issue other than to make a Google doc that lists all the biomes we are aware of, and keep it on this repository so it can be referenced and updated in the future. It would also be useful if more mods are added which add more biomes.

Edit2: OK, I've just discovered that the AT Launcher doesn't have the most up to date version of Unabridged 3 and that RWG has since been removed. The Unabridged subreddit still only references the AT Launcher, and I don't know how to get modpacks working from CurseForge yet, but I can probably figure it out. That's a separate issue, but I guess everything I've done so far is just a waste.
